packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3786,6 +3794,20 @@ export class TransactionController extends BaseController<
     this.#updateTransactionInternal(
```

Comment:
Not for this PR, but I wonder in retrospect if it would have been cleaner to encapsulate this update within the `GasFeePoller` itself via an `updateTransaction` callback. To keep the logic more modular and easier to test.